### PR TITLE
feat: add POST_INLINE_COMMENT MCP action

### PR DESCRIPTION
**Pull request by DGouron**
## Summary

- Adds `POST_INLINE_COMMENT` as a new MCP action type, enabling Claude to post inline comments on diffs without hardcoded `glab api`/`gh api` calls
- Introduces `DiffMetadata` (baseSha, headSha, startSha) pre-fetched by webhook controllers and passed through the review context — Claude never sees platform-specific SHAs
- Implements platform-specific CLI gateways (GitLab discussions API, GitHub pull comments API) with graceful degradation when diff metadata is unavailable
- Updates review skill templates to use MCP `POST_INLINE_COMMENT` instead of raw CLI commands

## Architecture

```
Entity Layer:
  reviewAction.schema.ts ─── POST_INLINE_COMMENT Zod schema
  diffMetadata.gateway.ts ── DiffMetadataFetchGateway interface
  reviewContext.ts ───────── DiffMetadata type + optional field

Interface Adapters:
  diffMetadataFetch.gitlab.gateway.ts ── glab API versions endpoint
  diffMetadataFetch.github.gateway.ts ── gh API pulls endpoint
  reviewAction.gitlab.cli.gateway.ts ─── glab discussions POST
  reviewAction.github.cli.gateway.ts ─── gh pulls/comments POST

MCP Layer:
  addAction.usecase.ts ────── validation + routing
  addAction.handler.ts ────── field extraction
  mcpServerStdio.ts ──────── tool definition update

Webhook Controllers:
  gitlab.controller.ts ────── pre-fetch diffMetadata (review + followup)
  github.controller.ts ────── pre-fetch diffMetadata (review)
  mrTrackingAdvanced.routes.ts ─ pre-fetch diffMetadata (manual followup)
```

## Test plan

- [x] 31 new/modified test cases across 8 test files (533/533 pass)
- [x] Schema validation: POST_INLINE_COMMENT Zod parsing (valid + invalid)
- [x] Guard integration: reviewActionGuard parses POST_INLINE_COMMENT
- [x] DiffMetadata gateways: GitLab versions API + GitHub pulls API parsing
- [x] CLI gateways: command building for both platforms + null when no diffMetadata
- [x] MCP usecase: validation rules (filePath required, line > 0, body required)
- [x] MCP handler: field extraction and routing
- [x] TypeScript, lint, and build pass clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/interface-adapters/gateways/diffMetadataFetch.gitlab.gateway.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+export class GitLabDiffMetadataFetchGateway implements DiffMetadataFetchGateway {
+  constructor(private readonly executor: CommandExecutor) {}
+
+  fetchDiffMetadata(projectPath: string, mergeRequestNumber: number): DiffMetadata {
+    const encodedProject = projectPath.replace(/\//g, '%2F')
+    const response = this.executor(
+      `glab api projects/${encodedProject}/merge_requests/${mergeRequestNumber}/versions`
+    )
+    const versions: GitLabMergeRequestVersion[] = JSON.parse(response)
+    const latest = versions[versions.length - 1]
```
Comment (DGouron) [inline src/interface-adapters/gateways/diffMetadataFetch.gitlab.gateway.ts]: ⚠️ **[IMPORTANT] No guard against empty versions array**

📍 `diffMetadataFetch.gitlab.gateway.ts:21-22`

`versions[versions.length - 1]` returns `undefined` when the array is empty. Accessing `.base_commit_sha` on `undefined` will throw a runtime error.

**Correction**: Add a guard before accessing the last element:
```ts
if (versions.length === 0) {
  throw new Error(`No diff versions found for MR ${mergeRequestNumber}`)
}
const latest = versions[versions.length - 1]
```

Also add a test for this edge case.
